Inferences/ForwardDemodulation.cpp

```diff
@@ -175,48 +179,13 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
           continue;
         }
 
-        // encompassing demodulation is fine when rewriting the smaller guy
-        if (toplevelCheck && _encompassing) {
-          // this will only run at most once;
```

Comment:
@mezpusz , could you please just quickly comment on why you moved this condition? (Not saying it's a bad idea, just curious about the reason.)

Reply:
Initially, I wanted to move everything that sets `toplevelCheck` into one line but that would have been messy. But still, setting it at the beginning seems clearer to me.

Not I get it why you ask: the ordering check is never run if there are no generalizations, which may save a bit of time, right? We can move it back, but then could we add a comment that it does not even run there necessarily, hence the place?

Reply:
We had a comment:

// this will only run at most once;
// could have been factored out of the getGeneralizations loop,
// but then it would run exactly once there

Still, happy to hear also @MichaelRawson. Not sure myself whether the benefit is super clear or simply worth the degradation of clarity.

Reply:
Sorry, I completely missed the "exactly" and "at most" part of the comment!

Reply:
I'm not sure anymore. The code _does_ get more readable by this PR regardless of this change, maybe we should put it back for now?

Reply:
I vote for shifting back. The diff will become smaller and it will be easier to visually check things are not getting more buggy than before :)

Reply:
I agree. Updated.